layers-samples/media-layer-sample.html

```diff
@@ -0,0 +1,204 @@
+              // Updates the button to start an XR session when clicked.
+              xrButton.addEventListener('click', onButtonClicked);
+              xrButton.disabled = video.readyState > 2;
+            }
+          });
+        }
+      }
+
+      // Called when the user clicks the button to enter XR. If we don't have a
+      // session we'll request one, and if we do have a session we'll end it.
+      function onButtonClicked() {
+        if (!xrSession) {
+          navigator.xr.requestSession('immersive-vr', {requiredFeatures: ['layers']}).then(onSessionStarted);
+          video.play();
+        } else {
+          xrSession.end();
+        }
+      }
+
+      // Called when we've successfully acquired a XRSession. In response we
+      // will set up the necessary session state and kick off the frame loop.
+      function onSessionStarted(session) {
+        xrSession = session;
+        xrButton.textContent = 'Exit VR';
+
+        // Listen for the sessions 'end' event so we can respond if the user
+        // or UA ends the session for any reason.
+        session.addEventListener('end', onSessionEnded);
+
+        xrMediaFactory = new XRMediaBinding(session);
+
+        session.requestReferenceSpace('local').then((refSpace) => {
+        xrRefSpace = refSpace;
+
+        let angle = 3.1415;
+        if (!video.src.includes("180"))
+          angle *= 2;
+
+        let layout = "stereo-top-bottom";
+        if (video.src.includes("sbs")||video.src.includes("SBS"))
+          layout = "stereo-left-right";
+
+        let layer = xrMediaFactory.createEquirectLayer(video, {space: refSpace, centralHorizontalAngle: angle, layout: layout});
+        // let layer = xrMediaFactory.createEquirectLayer(video, {space: refSpace});
+        session.updateRenderState({ layers: [ layer ] });
+
+          // Inform the session that we're ready to begin drawing.
+        session.requestAnimationFrame(onXRFrame);
+        });
+      }
+
+      // Called either when the user has explicitly ended the session by calling
+      // session.end() or when the UA has ended the session for any reason.
+      // At this point the session object is no longer usable and should be
+      // discarded.
+      function onSessionEnded(event) {
+        xrSession = null;
+        xrButton.textContent = 'Enter VR';
+        video.pause();
+      }
+
+      // Called every time the XRSession requests that a new frame be drawn.
+      function onXRFrame(time, frame) {
+        let session = frame.session;
+
+        // Inform the session that we're ready for the next frame.
+        session.requestAnimationFrame(onXRFrame);
+
+        for (let source of session.inputSources) {
+          if (source.gamepad) {
+            // Toggle Play/Pause on primary button press
+            if (buttonPressedThisFrame(source.gamepad, 0)) {
+              video.currentTime = 0;
+              video.play();
+            } else if (buttonPressedThisFrame(source.gamepad, 1)) {
```

Comment:
I was able to get the restart action to work consistently, but the squeeze action was unfortunately very flaky. I suspect it's because holding it for more than a frame will rapidly toggle between paused/unpaused and it's difficult for a human to hit exactly one frame's worth of input at 90Hz. Also, I initially thought that you'd have to hold the squeeze button and it would only remain paused while you did so. That might be a better model here?